eox_nelp/processors/xapi/event_transformers/grade_events.py

```diff
@@ -0,0 +1,69 @@
+"""
+Transformers for grading events.
+
+Classes:
+    SubsectionSubmittedTransformer: Transformer for the event nelc.eox_nelp.grades.subsection.submitted.
+"""
+from tincan import ActivityDefinition, LanguageMap, Result, Verb
+
+from eox_nelp.edxapp_wrapper.event_routing_backends import XApiTransformersRegistry, constants, event_transformers
+from eox_nelp.processors.xapi import constants as eox_nelp_constants
+
+
+@XApiTransformersRegistry.register("nelc.eox_nelp.grades.subsection.submitted")
+class SubsectionSubmittedTransformer(event_transformers.ProblemSubmittedTransformer):
+    """Transformer class for the event nelc.eox_nelp.grades.subsection.submitted."""
+
+    additional_fields = ('result', )
+
+    def get_verb(self):
+        """
+        Get verb for xAPI transformed event.
+
+        Returns:
+            `Verb`
+        """
+        return Verb(
+            id=constants.XAPI_VERB_ATTEMPTED,
+            display=LanguageMap({eox_nelp_constants.DEFAULT_LANGUAGE: constants.ATTEMPTED}),
+        )
+
+    def get_object_id(self):
+        """
+        Returns the object id.
+
+        Returns:
+            <str>: string representation of the block id.
+        """
+        return self.get_data("data.block_id")
+
+    def get_object_definition(self):
+        """
+        Returns the object definition.
+
+        Returns:
+            ActivityDefinition
+        """
+        return ActivityDefinition(
+            type=eox_nelp_constants.XAPI_ACTIVITY_UNIT_TEST
+        )
+
+    def get_result(self):
+        """
+        Get result for xAPI transformed event.
+
+        Returns:
+            `Result`
+        """
+        earned = self.get_data("data.earned") or 0
+        possible = self.get_data("data.possible") or 0
+
+        return Result(
+            success=earned >= possible,
```

Comment:
This success is only the boolean success for the subsection?

So knowing that... they don't need the percentage advance not only for the subsection, but also for the course?

https://github.com/eduNEXT/eox-nelp/blob/92b39a3ae989f5be890c598fbd16ef593e4ca459/eox_nelp/signals/receivers.py#L301

Reply:
yes, it's only for the subsection,

The scope of the event and the transformer is limited to subsections, this mustn't include any extra information in the result section